dev/reference: update transaction related documents (#1161)
pingcap/docs: dev/reference/transactions/transaction-model.md

**Note:**
>
> On the application side, remember to check the returned results of `COMMIT` because even there is no error in the execution, there might be errors in the `COMMIT` process.

## Differences from MySQL

### Transaction retry

> TiDB does **not** automatically retry failed transactions by default, because the retry can cause update loss. You can enable this feature by setting `tidb_disable_txn_auto_retry = 0`.

Reviewer: ```suggestion
While not enabled by default, TiDB can automatically retry failed transactions when `tidb_disable_txn_auto_retry = 0`. This feature is now disabled by default because retry may lead to lost updates.
```

+ Transaction

    When TiDB executes the `LOAD DATA` operation, a record with 20,000 rows of data is seen as a transaction for persistent storage by default. If one `LOAD DATA` operation inserts more than 20,000 rows, it is divided into multiple transactions to commit. If an error occurs in one transaction, this transaction in process is not committed. However, transactions before that are committed successfully. In this case, a part of the `LOAD DATA` operation is successfully inserted, and the rest of the data insertion fails. But MySQL treats a `LOAD DATA` operation as one transaction, one error leads to the failure of the entire `LOAD DATA` operation.

    > **Warning:**
    >

>     > The `LOAD DATA` operation in TiDB does not guarantee the atomicity of transactions, so it is not recommended to use it in production environment.

Reviewer: You can use it in production for loading data. You can't regularly use it once data is loaded.
```suggestion
    > The `LOAD DATA` operation in TiDB does not guarantee the atomicity of transactions, so it recommended only for initial batch loading data, and not during regular usage for production environment(s).
```
